mycroft/configuration/config.py

```diff
@@ -17,15 +17,15 @@
 import re
 import json
 import inflection
-from os.path import exists, isfile
+from os.path import exists, isfile, join, expanduser, dirname
```

Comment:
see previous comments on import organization and using `pathlib` instead of `os`

Reply:
is this an official requirement now?

Reply:
I think pathlib is often a good thing now that we don't need to support python 3.5 where pathlib was very wonky to work with. Pathlib sometimes makes code more convoluted though and is not the best in all circumstances IMO.

Anyway, regarding this PR: I don't think mixing in pathlib into code that is already using os.path functions extensively is desirable so a single call is now made using pathlib while the rest of the function uses os.path functions. Just makes it more of a jumble.

I think we can go over a couple of classes and functions and make them use pathlib instead of os.path in separate PRs to make the code more coherent.

Reply:
fair enough.  can we at least order the imports according to PEP8?

Reply:
Reordered